lib/chef/provider/user/mac.rb

```diff
@@ -0,0 +1,607 @@
+          # 2                                       How many properties we're going to set
+          # dsAttrTypeStandard:RecordName           Property 1: our users record name
+          # base64:dsAttrTypeNative:ShadowHashData  Property 2: our shadow hash data
+
+          import_file = ::File.join(Chef::Config["file_cache_path"], "#{new_resource.username}_password_dsimport")
+          ::File.open(import_file, "w+", 0600) do |f|
+            f.write <<~DSIMPORT
+              0x0A 0x5C 0x3A 0x2C dsRecTypeStandard:Users 2 dsAttrTypeStandard:RecordName base64:dsAttrTypeNative:ShadowHashData
+              #{new_resource.username}:#{::Base64.strict_encode64(shadow_hash_binary.string)}
+            DSIMPORT
+          end
+
+          run_dscl("delete", "/Users/#{new_resource.username}", "ShadowHashData")
+          run_dsimport(import_file, "/Local/Default", "M")
+          run_dscl("create", "/Users/#{new_resource.username}", "Password", "********")
+        ensure
+          ::File.delete(import_file) if defined?(import_file) && ::File.exist?(import_file)
+        end
+
+        def run_dsimport(*args)
+          shell_out!("dsimport", *(args.compact))
+        end
+
+        # Gets a value from user information hash using Chef attributes as keys.
+        #
+        def dscl_get(user_hash, key)
+          raise "Unknown dscl key #{key}" unless DSCL_PROPERTY_MAP.keys.include?(key)
+
+          user_hash[DSCL_PROPERTY_MAP[key]]
+        end
+
+        #
+        # Sets a value in user information hash using Chef attributes as keys.
+        #
+        def dscl_set(user_hash, key, value)
+          raise "Unknown dscl key #{key}" unless DSCL_PROPERTY_MAP.keys.include?(key)
+
+          user_hash[DSCL_PROPERTY_MAP[key]] = [ value ]
+          user_hash
+        end
+
+        def run_sysadminctl(args)
+          # sysadminctl doesn't exit with a non-zero code when errors are encountered
+          # and ouputs everything to STDERR instead of STDOUT and STDERR. Therefore we'll
+          # return the STDERR and let the caller handle it.
+          shell_out("sysadminctl", args).stderr
+        end
+
+        def run_dscl(*args)
+          result = shell_out("dscl", "-plist", ".", "-#{args[0]}", args[1..-1])
+          return "" if ( args.first =~ /^delete/ ) && ( result.exitstatus != 0 )
+          raise(Chef::Exceptions::DsclCommandFailed, "dscl error: #{result.inspect}") unless result.exitstatus == 0
+          raise(Chef::Exceptions::DsclCommandFailed, "dscl error: #{result.inspect}") if result.stdout =~ /No such key: /
+
+          result.stdout
+        end
+
+        def run_plutil(*args)
+          result = shell_out("plutil", "-#{args[0]}", args[1..-1])
+          raise(Chef::Exceptions::PlistUtilCommandFailed, "plutil error: #{result.inspect}") unless result.exitstatus == 0
+
+          if result.stdout.encoding == Encoding::ASCII_8BIT
+            result.stdout.encode("utf-8", "binary", undef: :replace, invalid: :replace, replace: "?")
```

Comment:
did this one actually happen? because chef-client does pretty well at forcing itself to utf-8 mode, and unless there was a truncated read (which there shouldn't be from shell_out) it should never return 8-bit encoding.

Reply:
I copy-pasta'd this from the `dscl` provider. Removing it things worked just fine.